### ISSUE TITLE
Gap between docs and code regarding transforms of pre-trained vision models

### ISSUE THREAD

**Issue by MaWentzien** [pytorch/vision#9425]
### 🐛 Describe the bug

Hi, 

can someone please help to clarify the following situation? 

The docs for pre-trained VGG16 (https://docs.pytorch.org/vision/stable/models/generated/torchvision.models.vgg16.html#torchvision.models.vgg16) and other models state 

> [...] Finally the values are first rescaled to [0.0, 1.0] and then normalized using [...]. 

However, looking at the code I see that the tensor is resized, cropped and normalized, but I do not see that the tensor is rescaled: 

The transformations are first defined here: https://github.com/pytorch/vision/blob/main/torchvision/models/vgg.py#L217-L222
```
 transforms=partial(
    ImageClassification,
    crop_size=224,
    mean=(0.48235, 0.45882, 0.40784),
    std=(1.0 / 255.0, 1.0 / 255.0, 1.0 / 255.0),
)
```

Then the ImageClassification transforms are executed here, if I am not mistaken: https://github.com/pytorch/vision/blob/main/torchvision/transforms/_presets.py#L58-L65

```
def forward(self, img: Tensor) -> Tensor:
    img = F.resize(img, self.resize_size, interpolation=self.interpolation, antialias=self.antialias)
    img = F.center_crop(img, self.crop_size)
    if not isinstance(img, Tensor):
        img = F.pil_to_tensor(img)
    img = F.convert_image_dtype(img, torch.float)
    img = F.normalize(img, mean=self.mean, std=self.std)
    return img
```

When is the actual rescaling happening?

Context: Currently I am using the VGG16_Weights.IMAGENET1K_FEATURES.transforms() for feature extraction. My image is scaled to [0,255]. The transformation scales the image to [-117, 65,000]. This results in VGG16 activations around 700,000 right before the classification head. If I manually scale to [0,1] the highest activation is around 236, which seems more reasonable to me.

Thank you for any hints.

### Versions

Applies to latest torchvision (0.25)

**Comment by NicolasHug** [comment]
Hi @MaWentzien , the rescaling of values happens in `convert_image_dtype`:

https://github.com/pytorch/vision/blob/b3c7d88ffe7a5a91d810e89b57b4cba4fa9d3b45/torchvision/transforms/functional.py#L216-L217

https://github.com/pytorch/vision/blob/b3c7d88ffe7a5a91d810e89b57b4cba4fa9d3b45/torchvision/transforms/_functional_tensor.py#L66

**Comment by MaWentzien** [comment]
Hi @NicolasHug, thank you a lot for point that out. That explains it and the way I fixed my code, seems correct so far. 

I was not looking in that function more deeply, as I did not expect, by the function name alone, that this function also scales my tensor during a safe conversion (int -> float). Especially there seems no need of doing so, as the scaling happens **after** converting it: https://github.com/pytorch/vision/blob/b3c7d88ffe7a5a91d810e89b57b4cba4fa9d3b45/torchvision/transforms/_functional_tensor.py#L98-L99 ?

In combination with the initial dtype() check, the scaling only happens conditionally. In my case the image tensor was already in the correct dtype and consequently not scaled, resulting in very high activations and my confusion.

From my user point of view, I would greatly appreciate to either give a small update in the docs (current docs suggest scaling happens **always**, which is not the case when your tensor is already in the correct dtype) or extract scaling into a separate function, when type conversion is safe without it. What do you think of it? What am I missing?

**Comment by NicolasHug** [comment]
> In my case the image tensor was already in the correct dtype

that's the problem. Float tensors should be in [0, 1] by convention. Behavior on any other range is undefined. That's a widely accepted convention, but unfortunately not one that is easily enforced to prevent such confusion.

https://docs.pytorch.org/vision/stable/transforms.html#dtype-and-expected-value-range

**Comment by NicolasHug** [comment]
> I was not looking in that function more deeply, as I did not expect, by the function name alone, that this function also scales my tensor during a safe conversion (int -> float)

This is a fair point and I agree that the name isn't obvious enough. That's why it's deprecated. In the v2 transforms, we recommend the more explicit `to_dtype(t, dtype=torch.float32, scale=True)` https://docs.pytorch.org/vision/main/generated/torchvision.transforms.v2.functional.to_dtype.html

**Comment by MaWentzien** [comment]
Thank you again. I was not aware of these links.
